### PR TITLE
Feat/persistent filters

### PR DESCRIPTION
**Pull request by martind260**
---
name: Pull request
about: Contribute your changes to the project
title: '[pull-request]'
labels: ''
assignees: ''
---
**Description**
Short description or comments

**Reference**
Issues https://app.zenhub.com/workspaces/digiwf-621f70bf50ea1100120b7e93/issues/it-at-m/digiwf-project/339

**Checklist**

- [x] Acceptance criteria are met
- [x] Pipeline has been run successfully
- [ ] JUnit tests have been run successfully
- [ ] Changelog is adjusted
- [ ] [Documentations](https://git.muenchen.de/groups/digitalisierung/-/wikis/Dokumentationen) are completed
- [x] Frontend is tested
- [ ] Created sub-branches are deleted
- [x] Boards are updated ( [Digitalisierung](https://git.muenchen.de/groups/digitalisierung/-/boards) , [Support](https://git.muenchen.de/digitalisierung/digiwf-support) , [Zenhub](https://app.zenhub.com/workspaces/digiwf-621f70bf50ea1100120b7e93/board) )
- [ ] feature process is created or extended
- [ ] [git-ops](https://git.muenchen.de/digitalisierung/digiwf-ops) is customized
- [ ] Openshift environments are prepared (Secrets, etc.)

### CODE REVIEW

File: digiwf-apps/packages/apps/digiwf-tasklist/src/components/task/TaskList.vue [hunk 1]
```diff
@@ -154,7 +177,71 @@ export default class TaskList extends Vue {
...
+  }
+
+  async savePersistentFilter() {
+    const request: SaveFilterTO = {
+      pageId: this.pageId,
+      filterString: this.syncedFilter,
+    }
+    try {
+      const cfg = ApiConfig.getAxiosConfig(FetchUtils.getPUTConfig({}));
+      await FilterRestControllerApiFactory(cfg).saveFilter(request);
```
Comment (lmoesle) [inline digiwf-apps/packages/apps/digiwf-tasklist/src/components/task/TaskList.vue]:
```suggestion
    const persistentFilter: SaveFilterTO = {
      pageId: this.pageId,
      filterString: this.syncedFilter,
    }
    try {
      const cfg = ApiConfig.getAxiosConfig(FetchUtils.getPUTConfig({}));
      await FilterRestControllerApiFactory(cfg).saveFilter(persistentFilter);
```

It is a bit misleading that the variable `request` is a `SaveFilterTO` object and not a http request. I would rename it to e.g. `persistentFilter`

File: digiwf-apps/packages/apps/digiwf-tasklist/src/views/ProcessInstances.vue [hunk 2]
```diff
@@ -163,5 +192,68 @@ export default class ProcessInstances extends Vue {
...
+  }
+
+  async loadPersistentFilters(refresh = false): Promise<void> {
+    this.persistentFilters = this.$store.getters['filters/filters'].filter((filter: FilterTO) => filter.pageId === "processinstances");
+    try {
+      await this.$store.dispatch('filters/getFilters', refresh);
+      this.errorMessage = "";
+    } catch (error) {
+      this.errorMessage = error.message;
+    }
```
Comment (lmoesle) [inline digiwf-apps/packages/apps/digiwf-tasklist/src/views/ProcessInstances.vue:250]: The code block with the api calls is exactly the same as in tasklist.vue. Should we move this into a seperate component or function?

File: digiwf-apps/packages/apps/digiwf-tasklist/src/views/Processes.vue [hunk 3]
```diff
@@ -132,6 +162,68 @@ export default class Processes extends Vue {
...
+
+  async loadPersistentFilters(refresh = false): Promise<void> {
+    this.persistentFilters = this.$store.getters['filters/filters'].filter((filter: FilterTO) => filter.pageId === "processes");
+    try {
+      await this.$store.dispatch('filters/getFilters', refresh);
+      this.errorMessage = "";
+    } catch (error) {
+      this.errorMessage = error.message;
+    }
+  }
```
Comment (lmoesle) [inline digiwf-apps/packages/apps/digiwf-tasklist/src/views/Processes.vue:221]: The code block with the api calls is exactly the same as in tasklist.vue and processinstance.vue. Should we move this into a seperate component or function?

**Review by lmoesle — APPROVED** [review]
👍🏻

File: digiwf-apps/packages/apps/digiwf-tasklist/src/views/ProcessInstances.vue (hunk 2 above)
Comment (martind260) [reply digiwf-apps/packages/apps/digiwf-tasklist/src/views/ProcessInstances.vue:250]: I thought about that. But the only solution at the time is a mixin, which isnt supported by composition api anymore. I recommend waiting until we switched all sources to composition api and then extract the api call logic to the composition variant of a mixin. Ive added an issue for that: https://github.com/it-at-m/digiwf-core/issues/46

File: digiwf-apps/packages/apps/digiwf-tasklist/src/views/Processes.vue (hunk 3 above)
Comment (martind260) [reply digiwf-apps/packages/apps/digiwf-tasklist/src/views/Processes.vue:221]: see above

File: digiwf-apps/packages/apps/digiwf-tasklist/src/views/ProcessInstances.vue (hunk 2 above)
Comment (lmoesle) [reply digiwf-apps/packages/apps/digiwf-tasklist/src/views/ProcessInstances.vue:250]: Sounds good to me. Thanks for opening the issue

**Pull request merged**